robottelo/hosts.py

```diff
@@ -479,6 +479,118 @@ def install_cockpit(self):
         if result.status != 0:
             raise ContentHostError('Failed to install the cockpit')
 
+    def register(
```

Comment:
How is this different from the `register_contenthost` function that's defined immediately belowm that uses subscription manager?

Reply:
This is using global registration api (post 6.10 way of registering hosts), register_contenthost and other functions here mimic the pre-6.10 manual way of registration. Some functions may still have value, but we should actually using globrex wherever possible (out of scope of this PR). 2 reasons for that:
- globrex starts to drift from the legacy method (e.g. it is not pulling keys in an rpm, so automation did hit https://github.com/SatelliteQE/robottelo/pull/9571 but customers didn't)
- new functionality is being added to globrex but not elsewhere. That's why I need to use this method -- `setup_remote_execution_pull` param enables pull client during registration

Reply:
@pondrejk ack, gotcha!

I'd like to discuss this further, and with @JacobCallahan make a plan for moving robottelo away from `register_contenthost` pattern.

Reply:
I did added support for generating the global registration command in Nailgun, Incase if that worthy to try https://github.com/SatelliteQE/nailgun/pull/795